src/app/context/context.component.ts

```diff
@@ -81,18 +81,20 @@ export class ContextComponent implements OnInit {
   constructor(
     public router: Router,
     private route: ActivatedRoute,
-    public dialog: MatDialog) {
+    public dialog: MatDialog,
+    private activityService: ActivityService) {
   }
 
   ngOnInit(): void {
-    // get the id from the current route
-    const routeParams = this.route.snapshot.paramMap;
-    const cardIdFromRoute = Number(routeParams.get('id'));
 
-    // find the deck for id we got from the route
-    this.activityData = allContextDecksCollection.find((array: { id: number; }) => Number(array.id) === cardIdFromRoute);
+    const activityId = Number(this.route.snapshot.url[1].path);
+
+    this.activityService.fetchContextActivity(activityId)
+      .subscribe((result) => {
+        this.activityData = result;
+        this.currentPack = this.activityData.cards;
+      });
 
-    this.pack = this.activityData.cards;
   }
 
   openDialog() {
```

Comment:
Formázás itt jól van? Csak. Mert mobilon rossznak látszik.

Reply:
Igen, mobil-probléma lehet. Így van:
        this.currentPack = this.activityData.cards;
      });
  }